File: src/scala/AreaActor.scala
package scala

import actors.Actor
import math.Pi

case class End()

case class Rectangle(height: Int, width: Int)

case class Circle(radius: Float)

class AreaActor extends Actor {
  def act() {
    loop {
      receive {
        case Rectangle(height, width) =>
          println("Area of the rectangle is " + (height * width))
        case Circle(radius) =>
          println(Pi * radius * radius)
        case End =>
          exit('ok)
        case n =>
          println("Can't calculate area of " + n)
      }
    }
  }
}

object AreaActor extends App {
  val areaActor = new AreaActor().start()
  areaActor ! Rectangle(22, 23)
  areaActor ! 'hello
  areaActor ! 22
  areaActor ! End

}

File: src/erlang/area_actor.erl
-module(area_actor).
-export([loop/0]).

loop() ->
	receive
		{rectangle, Width, Height} ->
			io:format("Area of rectangle is ~p~n", [Width*Height]),
			loop();
		{circle, Radius} ->
			io:format("Area of circle is ~p~n", [math:pi()*Radius*Radius]),
			loop();
		Other ->
			io:format("I do not know the area of ~p is ~n", Other),
			loop()
	end.
